packages/mds-stream-processor/tests/dead-letter.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { ValidationError } from '@mds-core/mds-utils'
+import { StreamProcessor } from '..'
+import { StreamSink } from '../@types'
+import { GeneratorSource } from '../connectors'
+
+// Updates a counter variable based on the number of writes.
+const CounterSink = (counter: number): StreamSink<string> => {
+  return () => ({
+    initialize: async () => undefined,
+    write: async () => {
+      counter++
+    },
+    shutdown: async () => undefined
+  })
+}
```

Comment:
🤦🏼 I realize now I should just do this with a jest mock and assert that it's called a certain amount of times, durr